app/models/invenio_rdm_record_converter.rb

```diff
@@ -145,58 +150,68 @@ def resource_type(digitalhub_subtype)
     end
   end
 
-  def creators(creators)
-    creators.map do |creator|
-      # Galter Health Sciences Library is a User on production DigitalHub, but it does NOT have a formal_name value
-      dh_user = User.find_by(formal_name: creator)
-
-      # Organization
-      if organization?(creator)
-        {
-          "person_or_org":
-            Hash.new.tap do |hash|
-              hash["name"] =  creator if creator.present? && (!creator.include?(CREATOR_NOT_IDENTIFIED) || !creator.include(CREATOR_UNKNOWN))
-              hash["type"] = "organisational"
-            end
-        }
-      # User within DigitalHub
-      elsif dh_user.present?
-        dh_user_formal_name = dh_user.formal_name.split(',') # split name into components to be reused
-        family_name = dh_user_formal_name.shift # remove first value from formal name
-        given_name = dh_user_formal_name.join(' ') # the remaining strings becomes given name
-
-        {
-          "person_or_org":
-            Hash.new.tap do |hash|
-              hash["type"] = "personal"
-              hash["given_name"] = given_name
-              hash["family_name"] = family_name
-              hash["identifiers"] = {"scheme": "orcid", "identifier": dh_user.orcid.split('/').pop} if dh_user.orcid.present?
-            end
-        }
-      # Unknown / Not Identified creator
-      elsif creator.downcase == CREATOR_UNKNOWN || creator.downcase == CREATOR_NOT_IDENTIFIED
-        {
-          "person_or_org":
-            Hash.new.tap do |hash|
-              hash["name"] = creator
-            end
-        }
-      # Personal record without user in database
-      else
-        family_name, given_name = creator.split(',')
-        {
-          "person_or_org":
-            Hash.new.tap do |hash|
-              hash["type"] = "personal"
-              hash["given_name"] = given_name.lstrip
-              hash["family_name"] = family_name.lstrip
-            end
-        }
-      end
+  def contributors(contributors)
+    contributors.map do |contributor|
+      contributor_json = build_creator_contributor_json(contributor)
+      contributor_json[:person_or_org].merge!({"role": ROLE_OTHER})
```

Comment:
Here is where the role is included for contributors per @fenekku's feedback.

Reply:
What happens if there are more than one contributors? Ruby will return each `contributor_json`, or just the last `contributor_json`?

Reply:
Just learned that`#map` returns the array. Thanks for explaining on the side @Meowcenary